pkg/chunked/storage_linux.go

```diff
@@ -2045,9 +2068,16 @@ func mustSkipFile(fileType compressedFileType, e internal.FileMetadata) bool {
 	return false
 }
 
-func (c *chunkedDiffer) mergeTocEntries(fileType compressedFileType, entries []internal.FileMetadata) ([]internal.FileMetadata, int64, error) {
+func (c *chunkedDiffer) mergeTocEntries(fileType compressedFileType, tocEntries []internal.FileMetadata) ([]fileMetadata, int64, error) {
 	var totalFilesSize int64
 
+	var entries []fileMetadata
+	for _, e := range tocEntries {
```

Comment:
AFAICS this copy/allocation would also be cheap to eliminate: We are already building `mergedEntries` as an array of the right type, so…